Header header
int16[] mode
uint32 num_switches